### ISSUE TITLE
NVIDIAPalgin (hsrminer-neoscrypt) doesnt benchmark with >1 gpu

### ISSUE THREAD

**Issue by Powaful3000** [MultiPoolMiner/MultiPoolMiner#1030]
When MPM is benchmarking it runs NVIDIAPalgin for the required 1 minute, with the miner working and reporting hashrate properly, but MPM doesn't read the hashrate properly and exempts it from the list of usable miners.  Editing the /Stats/PalginNvidia_NeoScrypt_HashRate.txt file and manually setting the hashrate is a workaround, however.

**Comment by UselessGuru** [comment]
> Maybe force 2-3 minute bench for this executable?

Up to you. This might help, but sometimes it will take even much longer to report the hashrates. :-(
Until Palgin offers a proper API it will not be officially be supported by MPM.

**Comment by papacheal** [comment]
Use at least "-interval 900" for a 15-minute run on benchmarks; this and many other miners take at least this long to warm up. For PalginNVidia, you'll also need to remove the "-watchdog" option, or MPM will kill it as if its dead when share submission slows down (again, because of the infrequent updates on aggregate hash rate).

Although is isn't officially supported, it's worth it as it rules the roost on NeoScrypt and HSR hashrate.

**Comment by UselessGuru** [comment]
There is another problem with Palgin.:
Because it takes ages to report hash rates MPM can also not update the stats in a timely manner, so the stats will age -> profit bias drops -> another miner will be activated.
There is no proper solution for this right now. Palgin needs to add a proper API.

**Comment by platinum4** [comment]
Tied to this as well: https://github.com/MultiPoolMiner/MultiPoolMiner/issues/991

**Comment by Nicheuji** [comment]
@UselessGuru @aaronsace nemosminer seemed to have found a solution. Could you review his codes and maybe apply the solution to MPM?

**Comment by UselessGuru** [comment]
> nemosminer seemed to have found a solution

Nope. I checked his solution. The problem is palgin not reporting overall hash rate within $interval.

**Comment by platinum4** [comment]
It didn't look any different except for forcing port 23333 and wrap $true

**Comment by Nicheuji** [comment]
Bummer :\
Anyway thanks for the clarification, guys.

**Comment by nemosminer** [comment]
nemosminer seemed to have found a solution

Nope. I checked his solution. The problem is palgin not reliably reporting overall hash rate within $interval.
There will be no real fix until palgin changes this, or adds a working API.

Yip.https://github.com/nemosminer/NemosMiner-v2.4.1/issues/198

**Comment by HaTaX2** [comment]
Something to add to this, I was curious how at one point my 1070 had pulled a benchmark score of 1.06MH/s while every time it was run it would close out early due to MPM not seeing any output.  I did some troubleshooting with .\Wrapper.ps1 and changed the line checker to include 'speed' in the list of words it looked for like so:
`if (($Line -like "*total*" -or $Line -like "*accepted*" -or $Line -like ">*" -or $Line -like "*speed*") -and $Line -like "*/s*") {`

My system was generating the speed reports often enough, but they were on different lines from 'accepted' so as I looked at it I thought it just wasn't catching the needed line / text.  This seems to have worked as it's been called up and run just fine since this change.

One other change I'll throw out there and this is more cosmetic then anything, but up at the top of Wrapper.ps1 around line 16 I've added:
`$host.ui.RawUI.WindowTitle = $FilePath`
This sets the title of the window to be that of the executed miner just as the unwrapped miners are.  I use a program that's called to manage the window location of the miners so I kinda needed this, but overall I think it improves the fit and finish some.
